### ISSUE TITLE
Case sensitivity check for xml documentation tags is wrong

### ISSUE THREAD

**Issue by MischkowskyM** [dotnet/roslyn#7822]
When compiling C# code, roslyn ignores all cases for XML Document tags. On the other hand, roslyn is case sensitive when processing VB code.

An example:

```
\\\ <Summary> Simple test method</summary>
\\\ <Param name="parameter">A sample parameter </Param>
\\\ <returns> The value of parameter</returns> 
public static bool TestMethod(bool parameter){ return parameter; }
```

This will compile fine in VS2015, but not VS2013. The documentation comments are also illegal for VB now. We discovered this, because a `<Param>`-tag inside our documentation caused the build to fail when we tested VS2015. We also believe that the implemented behavior is backwards to what it should be, as C# is usually case sensitive, while VB is not.

See file src/Compilers/Core/Portable/InternalUtilities/DocumentationCommentXmlNames.cs, the method ElementEquals(), line 40.

**Comment by AdamSpeight2008** [comment]
XML in VB is case-sensitive, as it predominately follow the language rules of XML. Case-insensitivity applies to identifiers.

**Comment by MischkowskyM** [comment]
Further information, as I finally found the time to test this myself in VB:

```
''' <Summary>
''' Defines the entry point of the application.
''' </Summary>
''' <Param name="Foo">
''' I am a non-existant param. I don't cause a warning, because I have an upper case P.
''' </Param>
''' <param name="Bar">
''' I am another non-existant param. I cause a warning: BC4207, as I should.
''' </param>
Sub Main()

End Sub
```
As the comments suggest, the first Param-tag does not cause any issues. It get's ignored like it used to be the case with C# in VS2013. The "param" tag causes a warning:
"Warning	BC42307	XML comment parameter 'Foo' does not match a parameter on the corresponding 'sub' statement".

For C#:

```
/// <summary>
/// Defines the entry point of the application.
/// </summary>
/// <Param name="Foo">
/// I am a non-existant param. I DO cause a warning (CS1572), even though I shouldn't.
/// </Param>
/// <param name="Bar">
/// I am another non-existant param. I also cause a warning (CS1572)
/// </param>
static void Main()
{

}
```
